### ISSUE TITLE
A proper cleanup is required when winutil removes onedrive , microsoft copilot and edge

### ISSUE THREAD

**Issue by MrDisguised** [ChrisTitusTech/winutil#2089]
**Describe the bug**

After removing one drive from my pc I realised that this script does indeed remove the onedrive application but there are many registry entries related to onedrive left that needs to be removed properly (a proper ""clean-up"" needs to be done). For example :- 

`HKEY_CURRENT_USER\Software\Microsoft\OneDrive`
`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\OneDrive`
`HKEY_CLASSES_ROOT\CLSID{018D5C66-4533-4307-9B53-224DE2ED1FE6}` : This registry key contains information related to the OneDrive shell extension.
`HKEY_CURRENT_USER\Software\Policies\Microsoft\OneDrive`
`HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\OneSyncSvc`
`HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\OneSyncSvc_365c8a`

.... And many more. You can open the `logon` tab of the `autoruns` application and you can see for yourself all the registry entries that needs to be removed. The Onedrive 1, Onedrive 2, Onedrive 3, Onedrive 4 , Onedrive 5 , Onedrive 6 and the Onedrive 7 registry folders.

**Expected behavior**
I expected that this script will remove all the registry entries after I remove the one drive application.

**Comment by MrDisguised** [comment]
I also think that the Microsoft edge and the Microsoft copilot registry entries should also be removed properly if someone removes them using this script. A proper clean-up is really necessary in my opinion.

**Comment by Marterich** [comment]
Definitely a fair argument. When implementing this, the undo action should be considered as well to restore a fully working system. (Maybe it's necessary to backup the keys somewhere to facilitate the restore option)
This will have to be investigated further

**Comment by MrDisguised** [comment]
> Definitely a fair argument. When implementing this, the undo action should be considered as well to restore a fully working system. (Maybe it's necessary to backup the keys somewhere to facilitate the restore option) This will have to be investigated further

I really don't have much idea about windows tbh lol. I am more of a linux guy but afaik winutil creates a system restore point by default before making changes to our pc. So if something goes wrong we can easily revert back to the old snapshot of our windows that was created by the winutil script.

I know it is **A LOT OF WORK**  to do but properly removing all the registry entries will make sure that the application (onedrive in this case) doesn't get installed again after some windows updates. Same goes with other applications like microsoft copilot and microsoft edge.

**Comment by Marterich** [comment]
Regarding Onedrive: 
I overhauled the uninstallation in a PR #2155 
Most of the Registry Keys and Startup Tasks are removed when the Onedrive Uninstallations finishes as designed (which will happen now more often I hope) The only key that was still there was `HKEY_CURRENT_USER\Software\Microsoft\OneDrive` I added this key to the cleanup section of the tweak. 

A thing to keep an eye out for is, under the above-mentioned Reg Key the DWORD UserInitiatedUninstall which was set to 1 after uninstallation. I'm not sure what this does or if this is needed to inhibit Windows of reinstalling the app. We'll have to keep this in mind and potentially add the value back at a later date. 

Regarding the Registry Keys:
`HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\OneSyncSvc`
`HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\OneSyncSvc_365c8a`
Both of them are also used by other programs on the device (like the mail app) so they are not directly OneDrive remnants.
